xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -818,7 +830,7 @@ unsigned int CAESinkAUDIOTRACK::AddPackets(uint8_t **data, unsigned int frames,
       if (m_passthrough && !m_info.m_wantsIECPassthrough)
       {
         if (written == size)
-          m_duration_written += m_format.m_streamInfo.GetDuration() / 1000;
+          m_duration_written += m_format.m_streamInfo.GetDuration(true) / 1000;
```

Comment:
This paplayer reuse here is somehow unclear. Don't we have a better chance to request the Duration? Does this mean: For IEC we still have the same amount of data, but basically more zeroes, therefore the time also is halfed. If we - in general now have less TrueHD playing time per buffer, this paplayer workaround could go away, right?

What's the reason paplayer is requesting "lower" amount of playing time?

Reply:
paplayer has nothing to do with here. The name of the parameter is unfortunate. It just means that it returns the duration of 12 audio units instead of 24.

Reply:
The RAW format has changed a lot since the TrueHD rework:

**Before:**
all sub-frames clipped to 2560 bytes and padded with zeros to 2560 so:

`2560 | 2560 | 2560  | 2560`.... up to 24      => size always fixed 2560 * 24 = 61440  (CBR)


**Now:**
sub-frames without padding and without clipping (any length):

`500 | 1250 | 30 | 2000 | 4200 | 800`.... up to 24    => size is variable (VBR)